src/pages/workspace/WorkspaceInitialPage.tsx

```diff
@@ -113,101 +109,97 @@ function WorkspaceInitialPage(props) {
                 announce = report;
             }
         });
+
         return [reports, admins, announce];
-    }, [policyID, props.reports]);
+    }, [policyID, reportsProp]);
 
-    /**
-     * Call the delete policy and hide the modal
-     */
+    /** Call the delete policy and hide the modal */
     const confirmDeleteAndHideModal = useCallback(() => {
-        Policy.deleteWorkspace(policyID, _.filter(policyReports, shouldArchiveReport), policy.name);
+        Policy.deleteWorkspace(policyID, policyReports.filter(shouldArchiveReport), policy?.name ?? '');
         setIsDeleteModalOpen(false);
         // Pop the deleted workspace page before opening workspace settings.
         Navigation.goBack(ROUTES.SETTINGS_WORKSPACES);
-    }, [policyID, policy.name, policyReports]);
+    }, [policyID, policy?.name, policyReports]);
 
     useEffect(() => {
-        const policyDraftId = lodashGet(props.policyDraft, 'id', null);
+        const policyDraftId = policyDraft?.id;
+
         if (!policyDraftId) {
             return;
         }
 
-        App.savePolicyDraftByNewWorkspace(props.policyDraft.id, props.policyDraft.name, '', props.policyDraft.makeMeAdmin);
+        App.savePolicyDraftByNewWorkspace(policyDraft.id, policyDraft.name, '', policyDraft.makeMeAdmin);
         // We only care when the component renders the first time
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
     useEffect(() => {
-        if (!isCurrencyModalOpen || policy.outputCurrency !== CONST.CURRENCY.USD) {
+        if (!isCurrencyModalOpen || policy?.outputCurrency !== CONST.CURRENCY.USD) {
             return;
         }
         setIsCurrencyModalOpen(false);
-    }, [policy.outputCurrency, isCurrencyModalOpen]);
+    }, [policy?.outputCurrency, isCurrencyModalOpen]);
 
-    /**
-     * Call update workspace currency and hide the modal
-     */
+    /** Call update workspace currency and hide the modal */
     const confirmCurrencyChangeAndHideModal = useCallback(() => {
-        Policy.updateGeneralSettings(policyID, policy.name, CONST.CURRENCY.USD);
+        Policy.updateGeneralSettings(policyID, policy?.name ?? '', CONST.CURRENCY.USD);
         setIsCurrencyModalOpen(false);
         ReimbursementAccount.navigateToBankAccountRoute(policyID);
-    }, [policyID, policy.name]);
+    }, [policyID, policy?.name]);
 
-    const policyName = lodashGet(policy, 'name', '');
-    const hasMembersError = PolicyUtils.hasPolicyMemberError(props.policyMembers);
-    const hasGeneralSettingsError = !_.isEmpty(lodashGet(policy, 'errorFields.generalSettings', {})) || !_.isEmpty(lodashGet(policy, 'errorFields.avatar', {}));
-    const hasCustomUnitsError = PolicyUtils.hasCustomUnitsError(policy);
-    const menuItems = [
+    const policyName = policy?.name ?? '';
```

Comment:
We could bring this variable to top of the component and use it in every place we have `policy?.name ?? ''`, wdyt?